Pinta/MainWindow.cs

```diff
@@ -393,17 +393,56 @@ private void MainWindow_DeleteEvent (object o, DeleteEventArgs args)
 
 		private void MainWindow_DragDataReceived (object o, DragDataReceivedArgs args)
 		{
+			// TODO: Generate random name for the picture being downloaded
+
 			// Only handle URIs
 			if (args.Info != 100)
 				return;
 
 			string fullData = System.Text.Encoding.UTF8.GetString (args.SelectionData.Data);
 
-			foreach (string individualFile in fullData.Split ('\n')) {
-				string file = individualFile.Trim ();
-
-				if (file.StartsWith ("file://"))
-					PintaCore.Workspace.OpenFile (new Uri (file).LocalPath);
+			try {
+				PintaCore.Workspace.HasOpenPendingDocuments = true;
```

Comment:
As far as I remember not setting this flag up could lead to something like NullPointerException or so in case of user trying to open more than one document at the same time.